File: translations/codenet/uncommented/Go2P/granite/Code/s803744887.py

import math

MX = 10000
sieve = [0] * MX
primes = []
for i in range(2, MX):
    if sieve[i]!= 0:
        continue
    primes.append(i)
    for j in range(i + i, MX, i):
        sieve[j] = i

n = int(input())
ans = 0
for _ in range(n):
    x = int(input())
    ans += 1
    for p in primes:
        if x > p and x % p == 0:
            ans -= 1
            break
print(ans)

File: translations/codenet/uncommented/Go2C/gpt-4o/Code/s803744887.c
#include <stdio.h>
#include <stdlib.h>

#define MX 10000

int main() {
    int sieve[MX] = {0};
    int *primes = malloc(MX * sizeof(int));
    int prime_count = 0;

    for (int i = 2; i < MX; i++) {
        if (sieve[i] != 0) {
            continue;
        }
        primes[prime_count++] = i;
        for (int j = i + i; j < MX; j += i) {
            sieve[j] = i;
        }
    }

    int n;
    scanf("%d", &n);
    int ans = 0;

    for (int i = 0; i < n; i++) {
        int x;
        scanf("%d", &x);
        ans++;
        for (int j = 0; j < prime_count; j++) {
            if (x > primes[j] && x % primes[j] == 0) {
                ans--;
                break;
            }
        }
    }

    printf("%d\n", ans);
    free(primes);
    return 0;
}

// <END-OF-CODE>
